frontend/lib/Repository/UserRepository/user_gateway.dart

```diff
@@ -16,17 +18,24 @@ class UserGateway {
 
   bool isAdmin() => user.userRoles.contains(UserRole.admin);
 
-  void refreshToken() async {
-    await authGetaway
-        .refreshToken()
-        .then((value) => user = User(value.tokens!));
-  }
-
   Future<AuthResponse> getFromMemory() async {
-    final String? access = await secureStorage.read(key: accessKey);
-    final String? refresh = await secureStorage.read(key: refreshKey);
+    String? access = await secureStorage.read(key: accessKey);
+    String? refresh = await secureStorage.read(key: refreshKey);
 
     if (access != null && refresh != null) {
+      if (!isTokenValid(access)) {
```

Comment:
[Sugestia] Można się pozbyć jednego if'a
```suggestion
    if (access != null && refresh != null && !isTokenValid(access)) {
```